ast/edit/option_editor.go

```diff
@@ -0,0 +1,153 @@
+package edit
+
+import (
+	"fmt"
+
+	"github.com/influxdata/flux/ast"
+	"github.com/influxdata/flux/semantic"
+	"github.com/influxdata/flux/values"
+)
+
+type EditFn func(opt *ast.OptionStatement) (ast.Expression, error)
+
+/*
+`Edit` passes the `OptionStatement` in the AST rooted at `node` that has the
+specified identifier to `editFn`.
+The function can have side effects on the option statement
+and/or return a non-nil `Expression` that is set as value for the option.
+If the value returned by the edit function is `nil` (or an error is returned) no new value is set
+for the option statement (but any, maybe partial, side effect is applied).
+`Edit` returns whether it could find and edit the option (possibly with errors) or not.
+*/
+func Edit(node ast.Node, optionIdentifier string, editFn EditFn) (bool, error) {
```

Comment:
nit, but the convention is to use `//` style comments even if they are large.

Reply:
Can you rename this function to `Option`? That way it reads like `edit.Option` when you use it and it lets us define other "edit" methods in the future.

Alternatively if you find that using a verb as the package name is confusing we could rename this package to `options` and then it becomes `options.Edit`. I am fine with either, so no as there is not stutter (i.e. edit.EditOptions), and option is somewhere in the name.

If you do rename the package to `options` then the `GetLiteral` function doesn't really belong in the package anymore and should be moved.